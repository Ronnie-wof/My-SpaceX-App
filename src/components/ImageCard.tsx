// Image card component. Used for rendering the image on rocket information screen.
import React, { useState } from 'react'
import {StyleSheet, View, Dimensions, Image, ActivityIndicator} from "react-native";
import { defaultImageUri } from '../configuration/spacexConfiguration';

const windowWidth = Dimensions.get('screen').width;
interface imageProps{
    uri: string
}

const ImageCard:React.FC<imageProps> = ({uri}) => {
    const [loading, setLoading] = useState(false);
    if(uri==null) {
        uri=defaultImageUri
    }
    return (
        <View style = {styles.cardContainer}>
            <Image source={{ uri: uri}} style={styles.image}
            resizeMode="contain"
            onLoadStart={() => setLoading(true)}
            onLoadEnd={() => setLoading(false)}
            />
            {loading && <ActivityIndicator color="green" size="large" />}
        </View>
    );
}

const radius = 20;
const styles = StyleSheet.create({
cardContainer: {
    width: windowWidth - 20,
    backgroundColor: '#faf0e6',
    borderRadius: radius,
    shadowColor: '#000000',
    shadowOffset: {
        width: 5,
        height: 5
    },
    shadowOpacity: 0.75,
    elevation: 9,
    marginVertical:10,
    marginHorizontal:10
},
image: {
    marginVertical:10,
    height:150,
    width: windowWidth - 25,
    borderTopLeftRadius: radius,
    borderTopRightRadius: radius,
    alignSelf: 'center'
},
});
export default ImageCard;