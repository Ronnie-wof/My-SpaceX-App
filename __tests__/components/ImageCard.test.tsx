// tests the ImageCard component
import React from 'react';
import renderer from 'react-test-renderer';
import ImageCard from '../../src/components/ImageCard';

describe('ImageCard component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ImageCard uri="https://images2.imgbox.com/48/a8/LTqq8OrE_o.png" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
