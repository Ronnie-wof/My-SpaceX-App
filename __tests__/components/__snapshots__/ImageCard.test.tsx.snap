// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#faf0e6",
      "borderRadius": 20,
      "elevation": 9,
      "marginHorizontal": 10,
      "marginVertical": 10,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.75,
      "width": 730,
    }
  }
>
  <Image
    onLoadEnd={[Function]}
    onLoadStart={[Function]}
    resizeMode="contain"
    source={
      {
        "uri": "https://images2.imgbox.com/48/a8/LTqq8OrE_o.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 150,
        "marginVertical": 10,
        "width": 725,
      }
    }
  />
</View>
`;
